### PR TITLE
Path: (WIP) Rewrite/refactor of postprocessor tests with minor refactor of PathPost.py

### PR DESCRIPTION
**Pull request by LarryWoestman**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [X]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [X]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [X]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [X]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [540352056](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/540352056) was triggered at [6a88120](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/6a881203d177365dbfb439f78468cb8a52260c14). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/PathScripts/PathPost.py [hunk 1]
```diff
@@ -135,6 +130,15 @@ def resolveFileName(self, job):
...
                 filename = filename.replace("%s", "")
+        return filename
+
+    def resolveFileName(self, job):
+        path = PathPreferences.defaultOutputFile()
+        if job.PostProcessorOutputFile:
+            path = job.PostProcessorOutputFile
+        filename = self.processFileNameSubstitutions(job, path)
+        if filename is None:
+            return None
```
Comment (benj5378) [inline src/Mod/Path/PathScripts/PathPost.py:141]: If filename is `None`, it should return `None` instead? Hummm, just `return None` should work then?

**Comment by LarryWoestman** [comment]
You only want to return None if the filename is None, you want to
continue to execute the rest of the code if the filename is not None.
Just returning None at this point would not produce the same result,
I believe.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [549189737](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/549189737) was triggered at [924d904](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/924d90460fe688c789b5cca2d7c8110e173e3207). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
You need to update your local master branch and then rebase your branch on to that. Then force push the changes to your github branch

**Comment by LarryWoestman** [comment]
I thought that is what I did :-)

I will check it out more carefully.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [549276044](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/549276044) was triggered at [0ec2930](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/0ec29309d29487bee4a920abee8967e72029311e). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by luzpaz** [comment]
Good job, although you still used `git merge` instead of `git rebase`

**Comment by LarryWoestman** [comment]
I think I am still confused.

Would you be willing to tell me the proper sequence of
commands to get the job done using the git command line?

What I am currently doing is:

cd testing
git pull --rebase upstream master
git pull origin testing  (to get my changes back in)
git push origin testing

Where "testing" is the branch that I have the pull request
for, "origin" is my Git repository, and "upstream" is the
"master" FreeCAD repository.

Thanks for all of the work that you do!

Larry

**Comment by luzpaz** [comment]
```
cd testing
git checkout master  # checkout the local master branch first 
git pull upstream master # pull upstream master changes in to your local git repo
git rebase upstream/master # rebase local master on upstream master changes
git checkout testing  # now checkout your local 'testing' branch 
git rebase upstream master  # Rebasing 'testing' branch on top of upstream master branch  
# make any changes to the files if you have any
# git add . # stage all changes
# git commit -m "WB-Name: succinct description of commit" -m "Body of commit where you can go in to more details"
git push --force origin testing 
```

**Comment by LarryWoestman** [comment]
Thank you!

Larry

**Comment by luzpaz** [comment]
You're welcome. We've all been there trying to decipher `git`. I'm still learning how to use it better. If you have any more questions, feel free to ask :wink:

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [550796782](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/550796782) was triggered at [ced3914](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/ced39143ea0dfb33458bc65605db8198e6d120bb). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [551625935](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/551625935) was triggered at [a81c225](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/a81c225aa392d10352dd1eb6a407034d0dfff0f8). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [552622767](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/552622767) was triggered at [2a39271](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/2a39271a6444afcd4c88c6cfb582545ffb2f36b0). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [553666578](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/553666578) was triggered at [594bfa4](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/594bfa4c2bd60ae8c9590d958f9d7da7659e9fc5). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [555583983](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/555583983) was triggered at [1db37ec](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/1db37ec40af352b9e9a0f1cd4c10b9302c924f5f). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [555828082](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/555828082) was triggered at [7524ced](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/7524ced4f337b29c3abbaf505a2808b56009597a). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [556114639](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/556114639) was triggered at [d5afc12](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/d5afc12dc0cfc02fe5c69f6ecbd333f5114e7690). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by sliptonic** [comment]
> ```shell
> cd testing
> git checkout master  # checkout the local master branch first 
> git pull upstream master # pull upstream master changes in to your local git repo
> git rebase upstream/master # rebase local master on upstream master changes
> git checkout testing  # now checkout your local 'testing' branch 
> git rebase upstream master  # Rebasing 'testing' branch on top of upstream master branch  
> # make any changes to the files if you have any
> # git add . # stage all changes
> # git commit -m "WB-Name: succinct description of commit" -m "Body of commit where you can go in to more details"
> git push --force origin testing 
> ```

Why are you rebasing master?
I usually keep my local master branch in sync with upstream/master and make all my changes in feature branches.  So if I had testing branch checked out, I can just do:
> git fetch --all
> git rebase upstream/master

To bring local master back in sync with upstream
> git fetch --all
> git checkout master
> git reset --hard upstream/master

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [559235210](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/559235210) was triggered at [d3db6ed](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/d3db6ed26deffff70e32ffd208e625c63a55da27). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [560241597](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/560241597) was triggered at [e48aee7](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/e48aee72ce9d6d26ac983eb00faa746205030e51). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894">pipeline status</a> for feature branch [PR_6894](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6894). Pipeline [561524809](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/561524809) was triggered at [2087cb0](https://github.com/FreeCAD/FreeCAD/pull/6894/commits/2087cb06e74ddc3b777ee04ed91b395939cbe55f). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request closed without merging**